src/devices/Shtc3/Shtc3.cs

```diff
@@ -0,0 +1,290 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Device;
+using System.Device.I2c;
+using System.Threading;
+using Iot.Units;
+
+namespace Iot.Device.Shtc3
+{
+    /// <summary>
+    /// Temperature and humidity sensor Shtc3
+    /// </summary>
+    public class Shtc3 : IDisposable
+    {
+        // CRC const
+        private const byte CRC_POLYNOMIAL = 0x31;
+        private const byte CRC_INIT = 0xFF;
+
+        /// <summary>
+        /// Default I2C address
+        /// </summary>
+        public const int I2cAddress = 0x70;
+
+        private I2cDevice _i2cDevice;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="Shtc3"/> class.
+        /// </summary>
+        /// <param name="i2cDevice">The I2C device used for communication.</param>
+        public Shtc3(I2cDevice i2cDevice)
+        {
+            _i2cDevice = i2cDevice;
+
+            Wakeup();
+            _status = Status.Idle;
+
+            Reset();
+        }
+
+        private Status _status;
+
+        /// <summary>
+        /// Set Shtc3 state
+        /// </summary>
+        public Status Status
+        {
+            get
+            {
+                return _status;
+            }
+            set
+            {
+                if (value == _status)
+                {
+                    return;
+                }
+
+                if (value == Status.Idle)
+                {
+                    Wakeup();
+                }
+                else
+                {
+                    Sleep();
+                }
+
+                _status = value;
+            }
+        }
+
+        /// <summary>
+        /// Try read Temperature and Humidity
+        /// </summary>
+        /// <param name="sensorMeasure">measure return by sensor</param>
+        /// <param name="LowPower">"true" measure in low power mode, "false"(default) measure in normal power mode</param>
+        /// <param name="ClockStretching">"true" allow clock stretching, "false"(default) without clock stretching</param>
+        /// <returns></returns>
+        public bool TryGetTempAndHumi(out Measure sensorMeasure, bool LowPower = false, bool ClockStretching = false)
```

Comment:
- no abbreviations in names (Humi => Humidity)
- the name suggests you will return two values and not structure consider just having two out args and remove the structure
- argument names should start with lower case (LowPower, ClockStretching => lowPower, clockStretching)